### PR TITLE
Units: Add scalable unit option

### LINKED ISSUE #50975: Unit / Stat: Unit selection not being used (autoscaling down)
**What happened**:
Unit format not respected after upgrading to Grafana 9.0
Unit is set to kWh (not Wh), but i.e. 140 Wh are displayed as "140 Wh".

image

**What you expected to happen**:
Display as 0.14 kWh as unit is set to kWh as on Grafana 8.x

**How to reproduce it (as minimally and precisely as possible)**:
Have value lower than 1000 with unit containing kilo

**Anything else we need to know?**:

**Environment**:
- Grafana version: 9.0
- Data source type & version:  InfluxDB 1.8.10-1
- OS Grafana is installed on: Raspbian 10.12
- User OS & Browser: Windows 10 // Chrome 102
- Grafana plugins: Default
- Others:

### PR DESCRIPTION
**Pull request by Develer**
**What is this feature?**

This PR provides scale option for unit standard edit option

**Which issue(s) does this PR fix?**:

Fixes #50975 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by axelavargas — COMMENTED** [review]
Hey @Develer 👋🏾 , great job tackling this PR, I gave it a test locally and it works as expected 🥳 .  

From my point of view is almost ready, would be great if users could a short description in the UI about this option. 

As well, we are missing a few things:

- Documentation regarding the new option, you can do that here: https://github.com/grafana/grafana/blob/main/docs/sources/panels-visualizations/configure-standard-options/index.md?plain=1#L57 
- Some unit test :) (we have `symbolFormatters.test.ts` and `valueFormats.test.ts`)

**Review by nmarrs — APPROVED** [review]
Added some docs and a few tests :) overall LGTM :) 

I really like how you did not change the default behavior and instead made this opt-in functionality - this helps us to prevent messing up users' dashboards 💯 


https://github.com/grafana/grafana/assets/22381771/8ad9b306-db8d-49b6-b42a-b27424318c50

**Review by imatwawana — COMMENTED** [review]
Thanks for the update to the docs. Two thoughts:

1. What's written there makes sense, but I don't understand the underlying logic of what triggers the scaling of units. Should we explain that logic a bit?
2. Since we're making this default behaviour, and will change the appearance of people's visualizations, I think it warrants a What's new entry.

File: packages/grafana-data/src/valueFormats/categories.ts [hunk 1]
```diff
@@ -274,52 +274,52 @@ export const getCategories = (): ValueFormatCategory[] => [
...
+      { name: 'kilogram (kg)', id: 'masskg', fn: SIPrefix('g', 1, scalable) },
       { name: 'metric ton (t)', id: 'masst', fn: toFixedUnit('t') },
     ],
   },
   {
     name: 'Length',
     formats: [
-      { name: 'millimeter (mm)', id: 'lengthmm', fn: SIPrefix('m', -1) },
+      { name: 'millimeter (mm)', id: 'lengthmm', fn: SIPrefix('m', -1, scalable) },
       { name: 'inch (in)', id: 'lengthin', fn: toFixedUnit('in') },
```
[the remark this answers is not in the document]
Comment (Develer) [reply packages/grafana-data/src/valueFormats/categories.ts:309]: yeah, indeed, right now it's only system international related units 🙂 We have to implement a separate scaling function for inches and feet.

**Comment by Develer** [comment]
> Thanks for the update to the docs. Two thoughts:
> 
> 1. What's written there makes sense, but I don't understand the underlying logic of what triggers the scaling of units. Should we explain that logic a bit?
> 2. Since we're making this default behaviour, and will change the appearance of people's visualizations, I think it warrants a What's new entry.

Before adding the scaling option, units have scaled by default, so there are no significant changes for the user. Technically, we only added the option to turn off scaling 😁

**Comment by imatwawana** [comment]
> > Thanks for the update to the docs. Two thoughts:
> > 
> > 1. What's written there makes sense, but I don't understand the underlying logic of what triggers the scaling of units. Should we explain that logic a bit?
> > 2. Since we're making this default behaviour, and will change the appearance of people's visualizations, I think it warrants a What's new entry.
> 
> Before adding the scaling option, units have scaled by default, so there are no significant changes for the user. Technically, we only added the option to turn off scaling 😁

Ah okay, I understand. Then it's fine as is.

**Review by imatwawana — APPROVED** [review]

**Pull request merged**